pycbc/waveform/bank.py

```diff
@@ -774,12 +774,15 @@ def generate_with_delta_f_and_max_freq(self, t_num, max_freq, delta_f,
         if cached_mem is None:
             wav_len = int(max_freq / delta_f) + 1
             cached_mem = zeros(wav_len, dtype=np.complex64)
-        if self.has_compressed_waveforms and self.enable_compressed_waveforms:
+
+        try :
+            if not (self.has_compressed_waveforms and self.enable_compressed_waveforms):
+                raise ValueError
```

Comment:
I'm a bit concerned that we could have a situation where `get_decompressed_waveform` isn't working, but we don't know because we're always falling back on the slower method.

Reply:
I've added a UserWarning so that if this happens, at least there is evidence, and the message of which includes the approximant as well, so that it is immediately obvious if something is wrong

Reply:
(Note that as the message changes, it will not be supressed for different approximants when repeated, see https://docs.python.org/3/library/warnings.html#repeated-warning-suppression-criteria)

Reply:
I meant that we should distinguish between `self.has_compressed_waveforms` being false, where we should silently fall back, and a call to `self.get_decompressed_waveform(` failing, which would indicate a genuine problem.

Reply:
(Not clear why `self.get_decompressed_waveform` failures would not be ValueErrors???)

Reply:
Ah, that makes sense. I've moved to a combined if/else with try/except which doesn't rely on the `get_decompressed_waveform` function not raising a `ValueError`